core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/keystore/persistence/KeyStoreRegistryPersistenceManager.java

```diff
@@ -0,0 +1,421 @@
+                    keyStoreName, e);
+        }
+    }
+
+    /**
+     * Generates an ID for the public certificate, which is used as a file name suffix for the certificate.
+     * e.g. If keystore name is 'example-com.jks', public cert name will be 'example-com-343743.cert'.
+     *
+     * @return generated id to be used as a file name appender.
+     */
+    private String generatePublicCertId() {
+
+        String uuid = UUIDGenerator.getUUID();
+        return uuid.substring(uuid.length() - 6, uuid.length() - 1);
+    }
+
+    /**
+     * Get the key store from the registry.
+     *
+     * @param keyStoreName Name of the key store.
+     * @param tenantId     Tenant Id.
+     * @return Key store.
+     * @throws SecurityException If an error occurs while getting the key store.
+     */
+    public KeyStore getKeyStore(String keyStoreName, int tenantId) throws SecurityException {
+
+        Registry registry = getGovernanceRegistry(tenantId);
+        if (!isKeyStoreExistsInRegistry(keyStoreName, registry)) {
+            throw new SecurityException("Key Store with a name: " + keyStoreName + " does not exist.");
+        }
+
+        try {
+            String path = RegistryResources.SecurityManagement.KEY_STORES + REGISTRY_PATH_SEPARATOR + keyStoreName;
+            Resource resource = registry.get(path);
+            byte[] bytes = (byte[]) resource.getContent();
+            KeyStore keyStore = KeystoreUtils.getKeystoreInstance(resource
+                    .getProperty(RegistryResources.SecurityManagement.PROP_TYPE));
+            char[] passwordChar = getKeyStorePasswordFromRegistryResource(resource, keyStoreName);
+            ByteArrayInputStream stream = new ByteArrayInputStream(bytes);
+            keyStore.load(stream, passwordChar);
+            resource.discard();
+            return keyStore;
+        } catch (RegistryException | KeyStoreException | NoSuchProviderException | IOException | CertificateException |
+                 NoSuchAlgorithmException e) {
+            throw new SecurityException("Error getting key store: " + keyStoreName, e);
+        }
+    }
+
+    /**
+     * Method to retrieve list of keystore metadata of all the keystores in a tenant.
+     *
+     * @param tenantId Tenant Id.
+     * @return List of KeyStoreMetaData objects.
+     * @throws SecurityException If an error occurs while retrieving the keystore data.
+     */
+    public List<KeyStoreMetadata> listKeyStores(int tenantId) throws SecurityException {
+
+        Registry registry = getGovernanceRegistry(tenantId);
+        List<KeyStoreMetadata> metadataList = new ArrayList<>();
+        try {
+            if (!registry.resourceExists(RegistryResources.SecurityManagement.KEY_STORES)) {
+                return metadataList;
+            }
+            Collection keyStoreCollection = (Collection) registry.get(RegistryResources.SecurityManagement.KEY_STORES);
+            String[] keyStoreFullNames = keyStoreCollection.getChildren();
```

Comment:
`keyStoreFullNames` -> using full name for the name is bit misleading, as it gives the impression there is a short name and there is a full name.. since/if there are no seperation like that, shall we keep a name like `keyStores`, `keyStoreNames`, `keyStorePaths`?

Reply:
`keyStorePaths` would be the best suited name given the context.

Reply:
Addressed with https://github.com/wso2/carbon-kernel/compare/6c4a7e5fc5efca69a114e5d3a08289b8dc809be7..3a9a8a70b486cbcffcd4d4da8c10588bd93f0358